### PR TITLE
Detail Collapsing Toolbar

### PR DESCRIPTION
**Pull request by BatmanInGotham2**
Added back and share buttons to detail toolbar. Collapses image on scroll with fade in and out animation. Uses scrollListener to determine when to show toolbar and when to use plant name in body as title of the page.

Screenshot_1562007263
Screenshot_1562007270

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt [hunk 1]
```diff
@@ -85,24 +136,29 @@ class PlantDetailFragment : Fragment() {
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
```
[the remark this answers is not in the document]
Comment (BatmanInGotham2) [reply app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt]: updated to remove

File: app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt [hunk 2]
```diff
@@ -19,13 +19,16 @@ package com.google.samples.apps.sunflower
 import android.content.Intent
 import android.os.Build
 import android.os.Bundle
+import android.util.Log
```
[the remark this answers is not in the document]
Comment (BatmanInGotham2) [reply app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt]: removed

File: app/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -37,9 +38,14 @@
     <dimen name="card_elevation">2dp</dimen>
     <dimen name="card_corner_radius">12dp</dimen>
 
+    <!-- elevation for Plant Detail toolbar - used to convert to pixels when hiding/showing -->
+    <dimen name="toolbar_elevation">5dp</dimen>
+
     <!-- Plant List Header margin -->
     <dimen name="header_margin">24dp</dimen>
 
+    <dimen name="plant_description_min_height">550dp</dimen>
```
Comment (tiembo) [inline app/src/main/res/values/dimens.xml]: Add comment explaining that this is to ensure that the top text can be fully scrolled off the screen

File: app/src/main/res/layout/fragment_plant_detail.xml [hunk 4]
```diff
@@ -90,6 +92,20 @@
...
+                    android:layout_height="wrap_content"
+                    android:layout_marginStart="@dimen/margin_small"
+                    android:layout_marginEnd="@dimen/margin_small"
+                    android:gravity="center_horizontal"
+                    android:text="@{viewModel.plant.name}"
+                    android:textAppearance="?attr/textAppearanceHeadline5"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    tools:text="Plant Name" />
```
Comment (tiembo) [inline app/src/main/res/layout/fragment_plant_detail.xml]: Change to actual plant name

Comment (BatmanInGotham2) [reply app/src/main/res/layout/fragment_plant_detail.xml]: updated

File: app/src/main/res/values/dimens.xml (hunk 3 above)
Comment (BatmanInGotham2) [reply app/src/main/res/values/dimens.xml]: updated

**Pull request merged**